### DISCUSSION TITLE
requirement for mountpoint

### DISCUSSION THREAD

**Discussion by n0xena** [openzfsonwindows/openzfs#420] [General]
so I was testing zfs the other day between linux (arch) and windows - and noticed: In order to get anything mounted on windows it requires a mountpoint to be set
mountpoint set to none (obviously) or legacy doesn't work
test setup looks quite simple:

pool - as root dataset - mountpoint set to none by zpool create and driveletter not touched
pool/set - as sole child dataset - mountpoint set to something useful for linux and driveletter to Z

it would be nice to have some workaround for a dataset with mountpoint set to legacy also be able to get mounted on windows - or is there something I miss?
the overall goal I would like to get done is a rather simple setup for removable media like thumbdrives (although exFAT seem to fit them better) or big external drives (by abusing a small FAT32 partition to provide the installer for systems without access to internet to download on thier own)

overall it works so far in a vm but for now I don't dare to try it on real hardware
I have high hopes in zfs to become a proper cross-platform filesystem - as other options are either FAT32 with all its limitations - or exFAT ... at least for exchange of data as trying to use it at least causes quite some issues on linux

keep up the good work

**Comment by lundman** [comment]
Basically, you are touching on the idea of setting mountpoint to legacy, then you can use the `mount` and `umount` commands to manually mount it as needed, except of course, Windows has no such commands. I can see we will perhaps add some simple executables to do just that at some point, or perhaps overload `zfs mount` with the option of passing a mountpoint for legacy mounts. 

Interestingly, `zfs mount` has a `-o option` but they missed having `mountpoint` as a valid temporary option, what a shame. Still it would be trivial to add `mountpoint` to be allowed, and use `zfs mount -o mountpoint=path pool/dataset`.

**Reply by n0xena** [reply]
I played around with zfs mount -o mountpoint - but as it doesn't give any fail stare back but seem to just ignore it this did not work
my real pool looks like this:
vault - root dataset, mountpoint none
vault/main - dataset for my user, mountpoint /home/main/vault (works great, btw)
a couple of zvols - basically backup images from the time before I moved from windows I still not yet completely reorganized into vault/main
so as my main dataset has a valid mountpoint I would be able to mount it by just setting the driveletter property on it - but due to the still quite unstable nature of zfsonwindows I just not dared to give it a shot - also as I run 2.3.0-rc3 and already executed zpool upgrade I can't test until zfsonwin also pushes to 2.3
nvm - I'll keep playin around in vms and maybe give it a try on bare metal with some old drives as test pool
